### ISSUE TITLE
Legal issues, license compatibility

### ISSUE THREAD

**Issue by kravemir** [mwanji/toml4j#56]
I found this project, and would like to use it. The project [depends on `com.google.gson.gson:gson`](https://mvnrepository.com/artifact/com.moandjiezana.toml/toml4j/0.7.2) library, [which is licensed under `Apache 2.0` license](https://mvnrepository.com/artifact/com.google.code.gson/gson/2.8.5), but `toml4j` is licensed under MIT license.

According to http://janelia-flyem.github.io/licenses.html, this is a legal issue, as `MIT` project can be included in `Apache 2.0` project, but not vice versa:

![image](https://user-images.githubusercontent.com/1643935/46490463-57046700-c808-11e8-95ca-bd2062003a66.png)

Or, my understanding of legal stuff around `MIT` and `Apache 2.0` is wrong,... However, both projects licensed under `Apache 2.0` is definitely a safe solution.

**Comment by mwanji** [comment]
Do you think there would be an issue in re-licensing to Apache 2.0?

The other option would be to remove Gson altogether, which I would eventually like to do, anyway.

**Comment by mwanji** [comment]
I just read the [Apache 2.0 license](https://www.apache.org/licenses/LICENSE-2.0) and do not think that there is currently a problem. It states in article 4: 

> You may add Your own copyright statement to Your modifications and may provide additional or different license terms and conditions for use, reproduction, or distribution of Your modifications, or for any such Derivative Works as a whole, provided Your use, reproduction, and distribution of the Work otherwise complies with the conditions stated in this License.

Also, in the link you posted, it says:

> This figure shows the rules when you are making other software part of the weakly protected component; there are other possibilities if you are only using the component as a library.

As I am using Gson as a library, I don't think the figure illustrates all possibilities.

While I cannot give legal advice, I do not think a legal issue has definitively been proven to exist.

**Comment by kravemir** [comment]
> Do you think there would be an issue in re-licensing to Apache 2.0?

Definitely, you can publish yours code under any license you want, even if you published it previously under different. As, you're the author, and license is just "agreement" between author and user, where user has to obligate terms, ie. license gives power to you.

If you have other contributors, then things are more complex,.. usually an explicit approval is needed, or removal of code authored by people not approving relicensing their contributions. There are also exceptions, and some licenses are compatbile to be relicensed without explicit approval, for example GPL to newer version of GPL.

> The other option would be to remove Gson altogether, which I would eventually like to do, anyway.

Why? I don't know gson very well, is there something wrong with it? Or just, you would like keep dependencies on absolute minimum?

> As I am using Gson as a library, I don't think the figure illustrates all possibilities.
>
> While I cannot give legal advice, I do not think a legal issue has definitively been proven to exist.

Probably legal issue doesn't exist. However, neither proved to not exist. For me, as an (potential) user of the library, it's a potential threat, that there might be some legal issues/problems in future. And, as I'm not good at legal laws/stuff, I tend to take safe ways to do things.

However, I don't see any else TOML library for Java. And, I like TOML's syntax more than JSON, XML, or any else I have yet found,...

**Comment by seekM** [comment]
> For me, as an (potential) user of the library, it's a potential threat, that there might be some legal issues/problems in future. And, as I'm not good at legal laws/stuff, I tend to take safe ways to do things.

I feel the same way. It would be cool if you could re-licensing your project to Apache 2.0 just to be on the safe side!
